rclcpp_components/src/node_main.cpp.in

```diff
@@ -33,41 +34,46 @@ int main(int argc, char * argv[])
   @executor@ exec;
   rclcpp::NodeOptions options;
   options.arguments(args);
-  std::vector<rclcpp_components::NodeInstanceWrapper> node_wrappers;
 
   std::string library_name = "@library_name@";
   std::string class_name = "rclcpp_components::NodeFactoryTemplate<@component@>";
 
   RCLCPP_DEBUG(logger, "Load library %s", library_name.c_str());
   auto loader = std::make_unique<class_loader::ClassLoader>(library_name);
   auto classes = loader->getAvailableClasses<rclcpp_components::NodeFactory>();
-  for (const auto & clazz : classes) {
-    std::string name = clazz.c_str();
-    if (name.compare(class_name) == 0) {
-      RCLCPP_DEBUG(logger, "Instantiate class %s", clazz.c_str());
-      std::shared_ptr<rclcpp_components::NodeFactory> node_factory = nullptr;
-      try {
-        node_factory = loader->createInstance<rclcpp_components::NodeFactory>(clazz);
-      } catch (const std::exception & ex) {
-        RCLCPP_ERROR(logger, "Failed to load library %s", ex.what());
-        return 1;
-      } catch (...) {
-        RCLCPP_ERROR(logger, "Failed to load library");
-        return 1;
-      }
-      auto wrapper = node_factory->create_node_instance(options);
-      auto node = wrapper.get_node_base_interface();
-      node_wrappers.push_back(wrapper);
-      exec.add_node(node);
-    }
+
+  if (std::find(
+    classes.begin(),
+    classes.end(),
+    class_name) == classes.end()) {
+    RCLCPP_ERROR(
+      logger,
+      "Class %s not found in library %s",
+      class_name.c_str(),
+      library_name.c_str());
+    return 1;
+  }
+  RCLCPP_DEBUG(logger, "Instantiate class %s", class_name.c_str());
+  std::shared_ptr<rclcpp_components::NodeFactory> node_factory = nullptr;
+  try {
+    node_factory = loader->createInstance<rclcpp_components::NodeFactory>(class_name);
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(logger, "Failed to load library %s", ex.what());
+    return 1;
+  } catch (...) {
+    RCLCPP_ERROR(logger, "Failed to load library");
+    return 1;
   }
+  // Scope to destruct node_wrapper before shutdown
```

Comment:
I removed the vector of node wrappers (since there was always just one) and in order to preserve the behavior where the node wrapper gets destructed before shutdown (in the case where spin doesn't block forever that is) I just added this small scope.

Reply:
> (since there was always just one)

Is that really the case?  It seems like this now changes the semantics.  That is, before we could load in multiple classes with the same name, and now it will do only one.

In other words; why did the `node_wrappers` vector exist?

Reply:
We can change this back if desired. I only changed it because de-nesting this code brought it to my attention since I got a compiler warning for variable shadowing.

Reply:
You are correct, this isn't doing anything as far as I can tell.  I believe it is a hold-over from when this was used as a standalone executable, and now it is a template that is populated by cmake, but wasn't changed to remove the nesting.

Reply:
OK, thanks for all of the feedback.  In that case I'm happy to remove the node_wrappers, I was just double-checking.